### PR TITLE
Sass styling

### PR DESCRIPTION
**Pull request by lpalbou**
This is a large refactoring of the Ribbon styling.

Some stylesheets have been renamed to follow BEM conventions. Because of that, even if the changes are not dramatic, I should probably handle the next merge with agr_ui repo, to check & verify everything is ok.

There is still some refactoring work to do to clean that up, but that's a good start.

Everything can be configured through the SASS variables in sass/_variables.scss

For backward compatibility, I have let the old index.css, however it should completely disappear in the integration with AGR. The demo is already using the main.scss.

You can configure the general appearance of the ribbon by changing the $ribbon-primary-color variable, and most other colors are derived from it. However, you have access to most relevant variables to configure more precisely the styling you want.

screen shot 2018-11-01 at 3 29 27 pm

### CODE REVIEW

File: src/config.js [hunk 1]
```diff
@@ -17,9 +17,9 @@ export function getConfig(mode) {
       'annot_color': '#3f51b5',
       'annot_url': getPhenoLinkTarget,
       'bio_link': BIOLINK + 'bioentityset/slimmer/phenotype?',
-      'evenRowColor': '#EAF0EF',
-      //'oddRowColor': '#FFFAE4',
-      'oddRowColor': '#ffffff',
+      // 'evenRowColor': '#EAF0EF',
```
Comment (nathandunn) [inline src/config.js]: can these just be removed?

File: src/dataHelpers.js [hunk 2]
```diff
@@ -230,6 +233,9 @@ function sortAssociations(assoc_a, assoc_b) {
 }
 
 export function heatColor(associations_count, hexColor, heatLevels) {
+//  console.log("heatColor: ", hexToRGB(variables.ribbon_strip_slim_saturation_color, 1));
```
Comment (nathandunn) [inline src/dataHelpers.js]: remove this line

File: src/sass/_base.scss [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+(-o-min-device-pixel-ratio: 3/2),
+(-webkit-min-device-pixel-ratio: 1.5),
+(min-device-pixel-ratio: 1.5),
+(min-resolution: 1.5dppx) {
+    .ontology-ribbon-species-icon_agr {
+        background-image: url(./icon/speciesSprite-2x.png);
+    }
+}
+
+// Doesn't seem to be used
```
Comment (nathandunn) [inline src/sass/_base.scss]: probably can just remove these

File: src/view/AssociationsView.js [hunk 4]
```diff
@@ -94,7 +94,9 @@ export default class AssociationsView extends React.Component {
             if (found) {
               shown++;            
             return (
-              <div className={'ontology-ribbon__assoc-row'} key={index} style={row_style} >
+
+              // <div className={'ontology-ribbon__assoc__row ' + (shown %2 == 0 ? ' ontology-ribbon__assoc__row--even ' : ' ontology-ribbon__assoc__row--odd ') + (focus ? ' ontology-ribbon__assoc__row--focus ' : ' ontology-ribbon__assoc__row--focus ' ) } key={index} >
```
Comment (nathandunn) [inline src/view/AssociationsView.js]: probably can remove

File: demo/src/Demo.js [hunk 5]
```diff
@@ -3,7 +3,9 @@ import PropTypes from 'prop-types';
 import { addUrlProps, UrlQueryParamTypes } from 'react-url-query';
 import { GridLoader } from 'react-spinners';
 
-import '../../src/index.css';
+// import '../../src/index.css';
```
Comment (nathandunn) [inline demo/src/Demo.js]: remove old comment

File: src/config.js (hunk 1 above)
Comment (lpalbou) [reply src/config.js]: Yes, I just let them in case while testing

**Comment by paaatrick** [comment]
I think the assignments in `_variables.scss` need to have the `!default` flag, right? Otherwise how would a client override them?

http://sass-lang.com/documentation/file.SASS_REFERENCE.html#variable_defaults_default

**Pull request merged**